docs/html/reference/pip_install.rst

```diff
@@ -402,28 +402,30 @@ Git
 ^^^
 
 pip currently supports cloning over ``git``, ``git+http``, ``git+https``,
-``git+ssh``, ``git+git`` and ``git+file``:
+``git+ssh``, ``git+git`` and ``git+file``,  but note that the ``git``, ``git+git``,
+and ``git+http`` are not recommended due to their lack of security.
```

Comment:
What is "their lack of security"? It'd help to be a bit more specific here (lack of TLS? insecure protocol? lack of any encryption?).

Further, I think we should be making this sentence into a [`.. warning::` directive](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-warning) instead, to better emphasize this, and draw attention to it.

Reply:
Changed to `Note that the git, git+git,and git+http are not recommended. (The former two use the Git Protocol. which lacks authentication, and HTTP is insecure due to lack of TLS based encryption)`